### ISSUE TITLE
Improved spotify integration

### ISSUE THREAD

**Issue by KraXen72** [bambanah/deemix#152]
**Describe the solution you'd like**
Hi! I had an idea for improving the spotify integration:
- [ ] improved spotify integration
	- [ ] separate spotify page
	- [ ] ability to download new songs from playlist after it's been changed
	- [ ] ability to sync playlist with spotify
		- if song filenames are numbered & playlists download to a separate folder, delete & re-download / only download missing & re-number
		- otherwise, re-link existing songs (resolve in library), download new ones and re-create m3u playlist.
	- [ ] generate a `.m3u` file pointing to relative songs 
	
**@bambanah, Is this something you'd merge / want in the app?**
This is something I'm even interested in eventually contributing (partially or fully), but no ETA, so if someone wants to do it before me, feel free to!

**Describe alternatives you've considered**
current implementation where spotify playlists are mixed with deezer ones in the favorites tab

**Additional context**
this would make for quite a convenient integration.

**Comment by bambanah** [comment]
Absolutely! I pretty much only use deemix to grab the latest songs I've added to spotify playlists, so better/more spotify integration is definitely something I'm keen for. Feel free to take a crack at it if you like, otherwise I'll try and get to this at some point in the future!
